deploy/awstools/awstools.py

```diff
@@ -220,17 +271,39 @@ def launch_instances(instancetype, count, instancemarket, spotinterruptionbehavi
                 ]),
                 "InstanceMarketOptions":marketconfig,
             }
-            if user_data_file is not None:
+            if user_data_file:
```

Comment:
I'm not a Python pro by any means, but wouldn't this error out in the `open`?

Reply:
This was just a nit that I gardened.  Usually `is not None` can be simplified to just using truthiness of the thing.  What are you worried about erroring out in the open call?

Reply:
Oh. Derp. Yes this is fine.